src/irr/asset/CImageLoaderKTX.cpp

```diff
@@ -0,0 +1,155 @@
+SOFTWARE.
+*/
+
+#include "CImageLoaderKTX.h"
+
+#ifdef _IRR_COMPILE_WITH_KTX_LOADER_
+
+#ifdef _IRR_COMPILE_WITH_GLI_
+#include "gli/gli.hpp"
+#endif
+
+#include "irr/asset/ICPUTexture.h"
+#include "irr/asset/CImageData.h"
+#include "COpenGLTexture.h"
+#include "CReadFile.h"
+#include "os.h"
+
+namespace irr
+{
+namespace asset
+{
+
+static bool checkKTXSignature(unsigned char* buffer)
+{
+    // Unique KTX identifier as per the Khronos KTX specification
+    // https://www.khronos.org/opengles/sdk/tools/KTX/file_format_spec/#2.1
+    unsigned char signature[12] = { 0xAB, 0x4B, 0x54, 0x58, 0x20, 0x31, 0x31, 0xBB, 0x0D, 0x0A, 0x1A, 0x0A };
+    
+    for (unsigned int i = 0; i < 12; i++)
+    {
+        if (buffer[i] != signature[i])
+            return false;
+    }
+    return true;
+}
+
+//! returns true if the file maybe is able to be loaded by this class
+bool CImageLoaderKTX::isALoadableFileFormat(io::IReadFile* _file) const
+{
+#ifdef _IRR_COMPILE_WITH_GLI_
+    if (!_file)
+        return false;
+
+    const size_t prevPos = _file->getPos();
+    
+    unsigned char buffer[12];
+
+    // Read the first few bytes of the KTX _file
+    if (_file->read(buffer, 12) != 12)
+    {
+        _file->seek(prevPos);
+        return false;
+    }
+
+    _file->seek(prevPos);
+
+    // Check if it really is a KTX _file
+    return checkKTXSignature(buffer);
+#else
+    return false;
+#endif // _IRR_COMPILE_WITH_GLI_
+}
+
+
+// load in the image data
+asset::SAssetBundle CImageLoaderKTX::loadAsset(
+    io::IReadFile*                                  _file,
+    const asset::IAssetLoader::SAssetLoadParams&    _params,
+    asset::IAssetLoader::IAssetLoaderOverride*      _override,
+    uint32_t                                        _hierarchyLevel)
+{
+    core::vector<asset::CImageData*> images;
+#ifdef _IRR_COMPILE_WITH_GLI_
+    if (!_file)
+        return {};
+
+    unsigned char buffer[12];
+
+    // Read the first few bytes of the KTX _file
+    if (_file->read(buffer, 12) != 12)
+    {
+        os::Printer::log("LOAD KTX: can't read _file\n", _file->getFileName().c_str(), ELL_ERROR);
+        return {};
+    }
+
+    // Check if it really is a KTX _file
+    if (!checkKTXSignature(buffer))
+    {
+        os::Printer::log("LOAD KTX: not really a ktx\n", _file->getFileName().c_str(), ELL_ERROR);
+        return {};
+    }
+
+    const char* filename = _file->getFileName().c_str();
+    gli::texture texture = gli::load_ktx(filename);
+    if (texture.empty())
+    {
+        os::Printer::log("Failed to load texture at path ", filename, ELL_ERROR);
+        return {};
+    }
+
+    // Instantiate translation class to convert GLI enums into OpenGL values
+    gli::gl GL(gli::gl::PROFILE_GL33);
+    const gli::gl::format textureFormat = GL.translate(texture.format(), texture.swizzles());
+    GLenum textureTarget = GL.translate(texture.target());
+
+    uint32_t nullOffset[3] = { 0, 0, 0 };
```

Comment:
if you want to support 1d,2d array textures or cubemaps you need to call this something else, and touch index 1 or 2 so you have the correct 3d extent

Reply:
I suggest to rename it to `minOffset`